pos_invoice_send_mail/models/pos_order.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2019 Druidoo - Iván Todorovich
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    @api.model
+    def create_from_ui(self, orders):
+        """ Inherit method to send by email """
+        order_ids = super().create_from_ui(orders)
+        refs_to_send_mail = [
+            o['data']['name'] for o in orders
+            if o['data'].get('to_send_mail')
+        ]
+        # Identify orders to send email
+        for order in self.browse(order_ids):
+            if (
+                order.config_id.iface_invoice_mail
+                and order.config_id.invoice_mail_template_id
```

Comment:
You could add a constraint (if iface is checked, template is required) and so remove this check.

Reply:
Implemented here: https://github.com/OCA/pos/pull/468/commits/8d299ec843cd039fa635f3b86e53391d8d1c31b0